### PR TITLE
hb2c group reduction update

### PR DESCRIPTION
**Pull request by Kvieta1990**
#### Summary of work

Small update in the reduction routine for HB2C (WAND^2) concerning the reduction of multiple datasets. The temperature value from the sample log will be included in the workspace title when using either `LoadWAND` or `WANDPowderReduction ` algorithm. This feature was requested by the HB2C team and it will benefit users in terms of better recognizing the reduced workspace title and its correspondence with the real-life variables (e.g., temperature).

### To test:

Just follow the test in the doc page for each algorithm to test it out.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/LoadWAND.py [hunk 1]
```diff
@@ -147,7 +147,15 @@ def PyExec(self):
             if len(runs) == 1:
                 RenameWorkspace("__tmp_load", outWS, EnableLogging=False)
             else:
-                outName = outWS + "_" + str(mtd["__tmp_load"].getRunNumber())
+                try:
+                    temp_val = mtd["__tmp_load"].run().getTimeAveragedValue("HB2C:SE:SampleTemp")
+                except RuntimeError as _:
+                    temp_val = 300.0
+
+                if temp_val == 0.0:
```
Comment (peterfpeterson) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/LoadWAND.py:155]: You may want to check for `nan` as well. I don't know if that would ever come out of the instrument.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/LoadWAND.py [hunk 2]
```diff
@@ -147,7 +147,15 @@ def PyExec(self):
             if len(runs) == 1:
                 RenameWorkspace("__tmp_load", outWS, EnableLogging=False)
             else:
-                outName = outWS + "_" + str(mtd["__tmp_load"].getRunNumber())
+                try:
+                    temp_val = mtd["__tmp_load"].run().getTimeAveragedValue("HB2C:SE:SampleTemp")
```
Comment (peterfpeterson) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/LoadWAND.py:151]: Do they not have the `SampleTemp` log alias setup in the DAS?

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py [hunk 3]
```diff
@@ -423,7 +423,18 @@ def _convert_data(self, input_workspaces):
         # such that all data can be binned exactly the same way.
 
         # BEGIN_FOR: located_global_xMin&xMax
-        output_workspaces = [f"{outname}{n+1}" for n in range(len(input_workspaces))]
+        output_workspaces = list()
+        for n, in_wksp in enumerate(input_workspaces):
+            try:
+                temp_val = mtd[in_wksp].run().getTimeAveragedValue("HB2C:SE:SampleTemp")
```
Comment (peterfpeterson) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py:429]: Same comments from `LoadWand`

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Question about the logname to use and minor changes to the try/except

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py (hunk 3 above)
Comment (Kvieta1990) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py:429]: same as above

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/LoadWAND.py (hunk 1 above)
Comment (Kvieta1990) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/LoadWAND.py]: added it in

**Review by peterfpeterson — COMMENTED** [review]
Adding the temperature name in should help keep track of things

**Review by peterfpeterson — APPROVED** [review]
I thought I approved this with my last comment. Oops

**Pull request merged**